File: task-03/Elixir/Sub-Task2.exs
defmodule FileCopy do
  # Main function to read from input file and write to output file
  def run do
    input_file = "input.txt"
    output_file = "output.txt"

    case File.read(input_file) do
      {:ok, content} ->
        # Write the content to output file
        case File.write(output_file, content) do
          :ok ->
            IO.puts("Content successfully copied from #{input_file} to #{output_file}.")
          {:error, reason} ->
            IO.puts("Failed to write to #{output_file}: #{reason}")
        end

      {:error, reason} ->
        IO.puts("Failed to read from #{input_file}: #{reason}")
    end
  end
end

# Run the program
FileCopy.run()

File: task-03/Rust/Sub-Task2.rs
use std::fs::File;
use std::io::{self, Read, Write};
use std::process;

fn main() {
    // Read the contents of input.txt
    let mut input_file = match File::open("input.txt") {
        Ok(file) => file,
        Err(error) => {
            eprintln!("Failed to open input.txt: {}", error);
            process::exit(1);
        }
    };

    let mut content = String::new();
    if let Err(error) = input_file.read_to_string(&mut content) {
        eprintln!("Failed to read from input.txt: {}", error);
        process::exit(1);
    }

    // Write the contents to output.txt
    let mut output_file = match File::create("output.txt") {
        Ok(file) => file,
        Err(error) => {
            eprintln!("Failed to create output.txt: {}", error);
            process::exit(1);
        }
    };

    if let Err(error) = output_file.write_all(content.as_bytes()) {
        eprintln!("Failed to write to output.txt: {}", error);
        process::exit(1);
    }

    println!("Content successfully copied from input.txt to output.txt");
}
